leptonai/cli/deployment.py

```diff
@@ -1299,9 +1321,46 @@ def update(
         spec=lepton_deployment_spec,
     )
 
+    if lepton_deployment.metadata.semantic_version:
+        dryrun_deployment = client.deployment.update(
+            name_or_deployment=name,
+            spec=new_lepton_deployment,
+            dryrun=True,
+        )
+
+        will_restart = not _same_major_version([
+            dryrun_deployment.metadata.semantic_version,
+            lepton_deployment.metadata.semantic_version,
+        ])
+        if will_restart:
+
+            confirmed = (not sys.stdin.isatty()) or Confirm.ask(
```

Comment:
@bobmayuze fyi - when this lands, it will be a user interface behavioral change. Here is a changelog string for you to use:

Deployment update cli will ask interactively whether the user wants to carry out the update if a rolling restart is needed. Previously, the cli will carry out update regardless of the need for rolling restart.

A rolling restart means that replicas will be restarted in a rolling fashion - every time a portion of new replicas with new parameters are started, and when they are ready, the same number of old replicas are terminated, and update carries on until all replicas are new. Rolling update is needed if any runtime environment, such as env variable, mounts, photon or container image version is changed. Simple updates, such as adding or removing access token, does not require rolling update.